src/components/implementation/fork_tests/malloc_comp/malloc_comp.c

```diff
@@ -0,0 +1,115 @@
+#include <print.h>
```

Comment:
(comment on the file name/location)

This directory structure would imply that this function exports an interface called `fork_tests`.  If this is not true, it should be in the `tests` directory that is an exception to the rule.

In the longer-term, we should likely have `unit_` and `micro_` "interface names" in the implementation directory for this stuff.